src/main/java/gift/application/token/apicaller/KakaoTokenApiCaller.java

```diff
@@ -0,0 +1,92 @@
+package gift.application.token.apicaller;
+
+import static io.jsonwebtoken.Header.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_FORM_URLENCODED_VALUE;
+
+import gift.global.config.KakaoProperties;
+import gift.global.validate.TimeOutException;
+import gift.model.token.KakaoToken;
+import java.net.URI;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.RequestEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class KakaoTokenApiCaller {
+
+    private final KakaoProperties kakaoProperties;
+    private final RestTemplate restTemplate;
+
+    public KakaoTokenApiCaller(KakaoProperties kakaoProperties, RestTemplate restTemplate) {
+        this.kakaoProperties = kakaoProperties;
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * 인가 코드를 사용해서 토큰을 가져옴
+     */
+    public KakaoToken getToken(String authorizationCode) {
+        var headers = createFormUrlencodedHttpHeaders();
+        var body = createGetAccessTokenBody(authorizationCode);
+
+        var request = new RequestEntity<>(body, headers, HttpMethod.POST,
+            URI.create(kakaoProperties.tokenRequestUri()));
+        try {
+            KakaoToken token = restTemplate.exchange(request, KakaoToken.class).getBody();
```

Comment:
이런 곳은 그냥 inline(Command + Option + N) 해버려도 괜찮습니다ㅎㅎ
```suggestion
            return restTemplate.exchange(request, KakaoToken.class).getBody();
```

Reply:
넵 반영하였습니다!